activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -184,12 +201,13 @@ def dump_to(filename)
       def marshal_dump
         reset_version!
 
-        [@version, @columns, {}, @primary_keys, @data_sources, @indexes, database_version]
+        [@version, @columns, {}, @primary_keys, @data_sources, @indexes, @foreign_keys, database_version]
       end
 
       def marshal_load(array)
-        @version, @columns, _columns_hash, @primary_keys, @data_sources, @indexes, @database_version = array
+        @version, @columns, _columns_hash, @primary_keys, @data_sources, @indexes, @foreign_keys, @database_version = array
```

Comment:
Note to self: I need to dig into what the backward and forward compatibility expectations are for marshaled schema cache.

I remember we want to be able to load a schema cache generated by the N-1 version to make it easy to upgrade, but I don't remember if that only applies to YAML serialization or if it applies to Marshal too.

Reply:
With the eager computing changes, I am thinking if we really need this schema cache changes anymore?

Reply:
It should apply for both. We want the cache to be read during the deploy of a new version, so that means that both the current version and the new one will try to read from the same file.